packages/extension-seo-url-mapping/src/setup-app.ts

```diff
@@ -0,0 +1,168 @@
+            (component: React.ComponentType<any>) =>
+                withApplicationExtensionStore(component, {
+                    id: extensionMeta.id,
+                    initializer: sliceInitializer
+                })
+        ]
+
+        return applyHOCs(App, HOCs)
+    }
+
+    /**
+     * This method is used to make changes to the PWA-Kit application routes. If your extension adds a new page to the application
+     * then you can add it to the router here. The routes passed to this method is an accrued list of routes that have been added
+     * from extensions applied before it. It is called during the `getRoutes` phase on both the server and the client.
+     *
+     * NOTE: If you instead want to modify a list of all the routes, refer to the `beforeRouteMatch` below.
+     */
+    extendRoutes(routes: RouteProps[]): RouteProps[] {
+        return routes
+    }
+
+    /**
+     * This method is used on the server during the rendering pipeline. It's provided a list of all the routes that your application
+     * is configured with, including those defined in the base application and those added by all the extensions. You can use this
+     * method to modify these routes in any way you want, but you must return an array of routes as a result.
+     */
+    beforeRouteMatch(allRoutes: RouteProps[]): RouteProps[] {
+        let configuredRoutes: RouteProps[] = []
+
+        const seoUrlMappingEnabled = true
+        // console.log('---- beforeRouteMatch CLIENT! allRoutes:')
+        // allRoutes.forEach((route: RouteProps) => {
+        //     console.log(route.path)
+        // })
+        console.log('---beforeRouteMatch isClient', !isServerSide)
+        if (!isServerSide) {
+            // CLIENT!
+            // Router Deserialization
+            const _routes = window.__CONFIG__.app.routes
+            // console.log('---- beforeRouteMatch CLIENT! window.__CONFIG__.app.routes:')
+            // _routes.forEach((route: RouteProps) => {
+            //     console.log(route)
+            // })
+            configuredRoutes = _routes.map(({path, componentName, componentProps}) => {
+                // DEVELOPER NOTE: We previously tried to dynamically load the component using the path to map to the
+                // filename and use import, but I couldn't get that to work. So here we are using the original routes
+                // array to find the component for a given path from the serialized route config. It doesn't completely
+                // work as it will remove the configured routes as they don't match the path. This should be done in
+                // another way.
+                // console.log('--- beforeRouteMatch in _routes.map:', path, 'componentName:', componentName, 'componentProps:', componentProps)
+                // TODO: Pass the components though the configuration
+                // - use a js file for configuration to import the files
+                const component = Pages[componentName]
```

Comment:
here is where we map the componentName to the actual component from `extension-chakra-storefront/src/pages`